src/components/shared/Tooltip.js

```diff
@@ -89,12 +89,14 @@ export default class Tooltip extends React.PureComponent<Props, State> {
     let offsetY = 0;
     if (interiorElement) {
       if (
-        mouseY + interiorElement.offsetHeight + MOUSE_OFFSET >
-        window.innerHeight
+        interiorElement.offsetHeight + MOUSE_OFFSET <
+        window.innerHeight - mouseY
       ) {
+        offsetY = 0;
+      } else if (interiorElement.offsetHeight + MOUSE_OFFSET < mouseY) {
         offsetY = interiorElement.offsetHeight + MOUSE_OFFSET;
       } else {
-        offsetY = -MOUSE_OFFSET;
+        offsetY = mouseY;
```

Comment:
I think the logic is not very easy to follow, because this logic defines `offsetY` , and then `offsetY` is used to calculate `top`. I think it would be easier to read if we'd calculate `top` directly in the `if` blocks. So it would be something like this:
```javascript
// By default, position the tooltip below the mouse cursor.
let top = mouseY + MOUSE_OFFSET;
if (interiorElement) {
  if (mouseY + MOUSE_OFFSET + interiorElement.offsetHeight >= window.innerHeight) {
    // if the tooltip doesn't fit below the mouse cursor
    if (mouseY - MOUSE_OFFSET - interiorElement.offsetHeight > 0) {
      // position the tooltip above the mouse cursor if it fits there
      top = mouseY - interiorElement.offsetHeight - MOUSE_OFFSET;
    } else {
      // otherwise, align the tooltip with the window's top.
      top = 0;
    }
  }

  // Now, do the same calculation for `left`
  ...
}

const style = {
  left,
  top
};
...
```

I also added some comments that make this code easier to read.

What do you think ?

Reply:
Hey @subashiniganesh, could you look at it? Do you need any more guidance? Thanks :)